test/swe2d/test_rossby_wave.py

```diff
@@ -0,0 +1,400 @@
+    c = Constant(modon_propagation_speed)
+    xi = x - c*t
+    psi = exp(-0.5*y*y)
+    phi = 0.771*(B/cosh(B*xi))**2
+    C = -0.395*B*B
+
+    # Zeroth order terms
+    eta_terms = phi*0.25*(3 + 6*y*y)*psi
+    if order == 0:
+        return interpolate(eta_terms, H_2d)
+
+    # Unnormalised Hermite series coefficients for eta
+    eta = np.zeros(28)
+    eta[0] = -3.0714300e+00
+    eta[2] = -0.3508384e-01
+    eta[4] = -0.1861060e-01
+    eta[6] = -0.2496364e-03
+    eta[8] = 0.1639537e-04
+    eta[10] = -0.4410177e-06
+    eta[12] = 0.8354759e-09
+    eta[14] = -0.1254222e-09
+    eta[16] = 0.1573519e-11
+    eta[18] = -0.1702300e-13
+    eta[20] = 0.1621976e-15
+    eta[22] = -0.1382304e-17
+    eta[24] = 0.1066277e-19
+    eta[26] = -0.1178252e-21
+
+    # Hermite polynomials
+    polynomials = [Constant(1.0), 2*y]
+    for i in range(2, 28):
+        polynomials.append(2*y*polynomials[i-1] - 2*(i-1)*polynomials[i-2])
+
+    # First order terms
+    eta_terms += C*phi*0.5625*(-5 + 2*y*y)*psi
+    eta_terms += phi*phi*psi*sum(eta[i]*polynomials[i] for i in range(28))
+
+    return interpolate(eta_terms, H_2d)
+
+
+def run(refinement_level, reference_solution, **model_options):
+    order = model_options.pop('expansion_order')
+    family = model_options.get('element_family')
+    model_comparison = model_options.pop('model_comparison')
+    overlap = model_options.pop('overlap')
+    stepper = model_options.get('timestepper_type')
+    print_output("--- running refinement level {:d} in {:s} space".format(refinement_level, family))
+
+    # Set up domain
+    lx, ly = 48, 24
+    nx, ny = 2*refinement_level, refinement_level
+    params = {'partition': True, 'overlap_type': (DistributedMeshOverlapType.VERTEX, overlap)}
+    mesh2d = PeriodicRectangleMesh(nx, ny, lx, ly, direction='x', distribution_parameters=params)
+    x, y = SpatialCoordinate(mesh2d)
+    mesh2d.coordinates.interpolate(as_vector([x-lx/2, y-ly/2]))
+
+    # Get simulation end time
+    T = model_options.get('simulation_end_time')
+    try:
+        assert T > 0.0 and T % 120 < 40.0
+    except AssertionError:  # TODO
+        raise NotImplementedError("Domain periodicity not accounted for in asymptotic expansion.")
+
+    # Physics
+    g = physical_constants['g_grav'].values()[0]
+    physical_constants['g_grav'].assign(1.0)
+    P1_2d = FunctionSpace(mesh2d, "CG", 1)
+    bathymetry2d = Function(P1_2d).assign(1.0)
+
+    # Create solver object
+    solver_obj = solver2d.FlowSolver2d(mesh2d, bathymetry2d)
+    options = solver_obj.options
+    options.timestepper_type = stepper
+    options.timestep = 0.96/refinement_level if model_comparison or stepper == 'SSPRK33' else 9.6/refinement_level
+    options.simulation_export_time = 5.0
+    options.simulation_end_time = T
+    options.use_grad_div_viscosity_term = False
+    options.use_grad_depth_viscosity_term = False
+    options.horizontal_viscosity = None
+    options.no_exports = True and refinement_level <= 48
```

Comment:
True and conditional = conditional. AFAIK you could drop the conditional as this gets overridden anyway